File: challenge/level1/소수 만들기.c
#include <stdio.h>

int is_prime(int num)
{
    for(int i = 2; i * i <= num; i++)
        if(!(num % i))
            return 0;

    return 1;
}

int solution(int nums[], size_t nums_len)
{
    int ans = 0;
    
    for(size_t i = 0; i < nums_len; i++)
        for(size_t j = i + 1; j < nums_len; j++)
            for(size_t k = j + 1; k < nums_len; k++)
                if(is_prime(nums[i] + nums[j] + nums[k]))
                    ans++;
    
    return ans;
}

File: study_KB/소수 만들기.java
class Solution {
    public int answer = 0;
    
    public boolean isPrime(int num) {
        if (num % 2 == 0)
            return false;
        if (num < 2)
            return false;
        for (int i = 3; i * i <= num; i += 2)
            if (num % i == 0)
                return false;
        return true;
    }
    
    public int solution(int[] nums) {
        dfs(nums, 0, 0, 0);
        return answer;
    }
    
    public void dfs(int[] nums, int num, int idx, int cnt) {
        if (cnt == 3) {
            answer = isPrime(num) == true ? answer + 1 : answer;
            return;
        }
        
        for (int i = idx; i < nums.length; i++)
            dfs(nums, num + nums[i], i + 1, cnt + 1);
    }
}
